rasa/importers/rasa.py

```diff
@@ -25,17 +26,37 @@ def __init__(
         domain_path: Optional[Text] = None,
         training_data_paths: Optional[Union[List[Text], Text]] = None,
     ):
-        if config_file and os.path.exists(config_file):
-            self.config = io_utils.read_config_file(config_file)
-        else:
-            self.config = {}
 
         self._domain_path = domain_path
 
         self._story_files, self._nlu_files = data.get_core_nlu_files(
             training_data_paths
         )
 
+        if config_file and os.path.exists(config_file):
+            config = io_utils.read_config_file(config_file)
+
+            # Would it be better to already check for missing config keys here, such
```

Comment:
I wouldn't optimize for performance at this point. The default case (hopefully) should be that users use Core + NLU anyway

Reply:
Yes, my point was rather this: For the smart configuration in future iterations, we need to load the training data and stories and pass them to the auto config. But if the user already specified `policies` in the `config.yml`, we only need the nlu training data to do auto configuration, not the stories. And if they specified the whole configuration, we don't need any training data. So we could avoid loading them here if we check beforehand which keys are specified.